From a89d5addd901de1c6086de23e16faa76aedad62e Mon Sep 17 00:00:00 2001
From: squimrel <squimrel@users.noreply.github.com>
Date: Sat, 15 Jul 2017 01:33:55 +0000
Subject: [PATCH 20/28] Fix for apple

Move iterators don't actually move ranges. They move every element in
the itererator range which is useless for chars.
---
 CMakeLists.txt    | 2 +-
 README.md         | 2 +-
 include/utility.h | 3 +--
 src/utility.cc    | 4 +---
 4 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6d74919..63ea6e1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.0)
 # Don't confuse it with the libcdio-devel which provides a libiso9660.so.
 project(iso9660io)
 
diff --git a/README.md b/README.md
index 3500cae..7b03f35 100644
--- a/README.md
+++ b/README.md
@@ -7,7 +7,7 @@ A basic and incomplete implementation of ECMA-119.
 ```
 mkdir -p build/
 cd build/
-cmake -DMAKE_INSTALL_PREFIX:PATH=$PWD ..
+cmake -DCMAKE_INSTALL_PREFIX:PATH=$PWD ..
 make
 make install
 ```
diff --git a/include/utility.h b/include/utility.h
index 6731403..7b6f7e6 100644
--- a/include/utility.h
+++ b/include/utility.h
@@ -62,8 +62,7 @@ T integer(iso9660::Buffer::const_iterator first,
 
   std::size_t i = 0;
   return std::accumulate(
-      std::make_move_iterator(first), std::make_move_iterator(last),
-      static_cast<T>(0),
+      first, last, static_cast<T>(0),
       [&i, bits](T result, iso9660::Buffer::value_type byte) {
         result |= static_cast<T>(byte) << (bits - i * BITS_IN_BYTE);
         ++i;
diff --git a/src/utility.cc b/src/utility.cc
index 9e11496..f80a1ee 100644
--- a/src/utility.cc
+++ b/src/utility.cc
@@ -72,9 +72,7 @@ void utility::big_endian(std::string* const raw) {
 std::string utility::substr(iso9660::Buffer::const_iterator first,
                             iso9660::Buffer::const_iterator last,
                             std::size_t at, std::size_t size) {
-  return std::string(
-      std::make_move_iterator(first + at),
-      std::make_move_iterator(size == 0 ? last : (first + at + size)));
+  return std::string(first + at, size == 0 ? last : (first + at + size));
 }
 
 iso9660::Buffer::value_type utility::at(iso9660::Buffer::const_iterator first,
-- 
2.14.1

